File: prob3_bronze_season26contest1/python/single_test_runner.py
import sys
from pathlib import Path

import solution


def main():
    default_path = Path(__file__).with_name("single_test_input.txt")
    path = Path(sys.argv[1]) if len(sys.argv) >= 2 else default_path
    if not path.is_file():
        print("Input file not found:", path)
        if path == default_path:
            print("Create it and paste input like: 4 2 3 2 2 11 3 4 3 3 1 100")
        return

    tokens = path.read_text(encoding="utf-8").strip().split()
    if not tokens:
        print("Input file is empty:", path)
        return

    if len(tokens) < 5:
        print("Input file does not contain enough values:", path)
        return

    n = int(tokens[0])
    k = int(tokens[1])
    if len(tokens) >= 3:
        q = int(tokens[2])
        if len(tokens) == 3 + q * 3:
            idx = 3
            updates = []
            for _ in range(q):
                r = int(tokens[idx]) - 1
                c = int(tokens[idx + 1]) - 1
                v = int(tokens[idx + 2])
                idx += 3
                updates.append((r, c, v))
        elif (len(tokens) - 2) % 3 == 0:
            updates = []
            idx = 2
            while idx + 2 < len(tokens):
                r = int(tokens[idx]) - 1
                c = int(tokens[idx + 1]) - 1
                v = int(tokens[idx + 2])
                idx += 3
                updates.append((r, c, v))
        else:
            print("Input does not match expected format (N K Q followed by Q updates).")
            return
    else:
        updates = []

    for value in solution.solve_all(n, k, updates):
        print(value)


if __name__ == "__main__":
    main()

File: prob3_bronze_season26contest1/cpp/single_test_runner.cpp
#define INCLUDED_IN_TEST
#include "solution.cpp"
#include <fstream>
#include <iostream>
#include <string>
#include <tuple>
#include <vector>
using namespace std;

static vector<long long> read_numbers(const string &path) {
    ifstream in(path);
    vector<long long> values;
    long long x;
    while (in >> x) {
        values.push_back(x);
    }
    return values;
}

int main(int argc, char **argv) {
    string path = (argc > 1) ? argv[1] : string("single_test_input.txt");
    ifstream check(path);
    if (!check.good()) {
        cout << "Input file not found: " << path << "\n";
        if (path == "single_test_input.txt") {
            cout << "Create it and paste input like: 4 2 3 2 2 11 3 4 3 3 1 100\n";
        }
        return 0;
    }

    vector<long long> values = read_numbers(path);
    if (values.size() < 5) {
        cout << "Input file does not contain enough values: " << path << "\n";
        return 0;
    }

    int n = static_cast<int>(values[0]);
    int k = static_cast<int>(values[1]);

    vector<tuple<int, int, ll>> updates;
    if (values.size() >= 3) {
        long long q = values[2];
        size_t expected = 3 + static_cast<size_t>(q) * 3;
        if (values.size() == expected) {
            updates.reserve(static_cast<size_t>(q));
            size_t idx = 3;
            for (long long i = 0; i < q; i++) {
                int r = static_cast<int>(values[idx++]) - 1;
                int c = static_cast<int>(values[idx++]) - 1;
                ll v = values[idx++];
                updates.emplace_back(r, c, v);
            }
        } else if ((values.size() - 2) % 3 == 0) {
            long long inferred = static_cast<long long>((values.size() - 2) / 3);
            updates.reserve(static_cast<size_t>(inferred));
            size_t idx = 2;
            for (long long i = 0; i < inferred; i++) {
                int r = static_cast<int>(values[idx++]) - 1;
                int c = static_cast<int>(values[idx++]) - 1;
                ll v = values[idx++];
                updates.emplace_back(r, c, v);
            }
        } else {
            cout << "Input does not match expected format (N K Q followed by Q updates).\n";
            return 0;
        }
    }

    vector<ll> outputs = solve_all(n, k, updates);
    for (ll value : outputs) {
        cout << value << "\n";
    }
    return 0;
}
